grafana-plugin/src/containers/ColumnsSelectorWrapper/ColumnsModal.tsx

```diff
@@ -0,0 +1,228 @@
+                        <Block bordered withBackground fullWidth className={styles.valuesBlock}>
+                          {result.values === undefined ? (
+                            <LoadingPlaceholder text="Loading..." className="loadingPlaceholder" />
+                          ) : (
+                            renderLabelValues(result.name, result.values)
+                          )}
+                        </Block>
+                      )}
+                    </VerticalGroup>
+                  ))}
+                </VerticalGroup>
+              )}
+
+              {inputRef?.current?.value && searchResults.length === 0 && (
+                <Text type="primary">0 results for your search.</Text>
+              )}
+            </VerticalGroup>
+          </div>
+
+          <HorizontalGroup justify="flex-end" spacing="md">
+            <Button variant="secondary" onClick={onCloseModal}>
+              Close
+            </Button>
+            <WithPermissionControlTooltip userAction={UserActions.OtherSettingsWrite}>
+              <Button
+                disabled={isLoading || !searchResults.find((it) => it.isChecked)}
+                variant="primary"
+                onClick={onAddNewColumns}
+              >
+                {isLoading ? <LoadingPlaceholder className={'loader'} text="Loading..." /> : 'Add'}
+              </Button>
+            </WithPermissionControlTooltip>
+          </HorizontalGroup>
+        </VerticalGroup>
+      </Modal>
+    );
+
+    function renderLabelValues(keyName: string, values: LabelValue[]) {
+      return (
+        <HorizontalGroup spacing="xs">
+          {values.slice(0, 2).map((val) => (
+            <LabelTag label={keyName} value={val.name} key={val.id} />
+          ))}
+          <div className={styles.totalValuesCount}>{values.length > 2 ? `+ ${values.length - 2}` : ``}</div>
+        </HorizontalGroup>
+      );
+    }
+
+    async function expandOrCollapseSearchResultItem(result: SearchResult, index: number) {
+      setSearchResults((items) =>
+        items.map((it, idx) => (idx === index ? { ...it, isCollapsed: !it.isCollapsed } : it))
+      );
+
+      await fetchLabelValues(result, index);
+    }
+
+    async function fetchLabelValues(result: SearchResult, index: number) {
+      const labelResponse = await store.alertGroupStore.loadValuesForLabelKey(result.id);
+
+      setSearchResults((items) =>
+        items.map((it, idx) => (idx === index ? { ...it, values: labelResponse.values } : it))
+      );
+    }
+
+    function onCloseModal() {
+      inputRef.current.value = '';
+
+      setSearchResults([]);
+      setIsModalOpen(false);
+      setTimeout(forceOpenToggletip, 0);
+    }
+
+    async function onAddNewColumns() {
+      const mergedColumns = [
+        ...store.alertGroupStore.columns,
+        ...searchResults
+          .filter((item) => item.isChecked)
+          .map(
+            (it): AGColumn => ({
+              id: it.id,
+              name: it.name,
```

Comment:
Small thing and general notice. It's a good practice to use more explicit names and prevent such shortcuts. Here we're iterating over "results", then in filter we call a single entry an "item" and then an "it". Maybe we can call it a "result"?